### DISCUSSION TITLE
Problems with connection

### DISCUSSION THREAD

**Discussion by YanaS123** [spine-tools/SpineOpt.jl#1032] [Support]
Hello! I have some problems with "connection". I created node "Storage node" for accumulating energy and node "Electricity node" where energy is consumed. After I tried to connect these nodes between each other with "connection" but it didn't work. Unfortunately I can't see how node "Storage node" accumulates energy and transmit it to "Electricity node". And I don't understand there are problems with parameters for "connection" or parameters for another blocks of model. 

I attach the image of the model and the model itself in zip format.

Thanks in advance!

![Model with connection](https://github.com/spine-tools/SpineOpt.jl/assets/164343271/2118bc85-1a59-4e3e-8c55-8954a3db9430)
[Project with connection.zip](https://github.com/user-attachments/files/15572716/Project.with.connection.zip)

**Comment by tarskul** [comment]
Hi!

You did not specify the version of spine tools that you are using so I could not open the attached sqlite database in the latest version.

It is also hard for me to tell what exactly the problem is. Can you be a bit more specific (and isolate the problem)?

But here are some general tips:
- Several power flows can arrive at a node. Those powerflows need to be balanced. You have several balancing options. Including 'balance type' set to None if you want the node to act like a source and 'balance type' set to node if you want the sum of the flows to be zero. Additionally a node can act like a sink when the demand has been specified and can help balance the flows with storage. Have your parameters been set accordingly?
- Typically, the best approach for dealing with spine models is to build them up step by step. For example, start with the electricity node, the coal node and the coal plant and make sure that runs first. That will also make it easier to isolate the problem.

**Reply by YanaS123** [reply]
Thank you for your tips! I use 0.1.0.dev0 version of Spine Toolbox. 
Unfortunately I have little expirience using in SpineOpt and I don't understand how to create connection between two nodes (in my case "storage node" and "electricity") and what blocks I have to use for this. I used "connection" but maybe there is another methods to creat  this and I don't need "connection".

**Comment by DillonJ** [comment]
Hi @YanaS123,

To compliment what @tarskul has said - the `has_state` parameter determines whether or not a node has storage. If it has the value true, then the node will have a storage variable. The `node` parameter `node_state_cap` is the storage capacity of the node. I think you can accomplish what you want simply by specifing `has_state` = true for the electricity node without the need for another node altogether. However, if you want to put capacities on the transfers to and from the storage, then you will need a second node with `has_state` = true and use units or connections (depending on what you are trying to represent) to specify capacities on the flows.

**Reply by YanaS123** [reply]
I see what you mean. Thank you for your reply!
